subsys/bluetooth/host/classic/hfp_ag.c

```diff
@@ -0,0 +1,2106 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <errno.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/printk.h>
+
+#include <zephyr/bluetooth/conn.h>
+
+#include "common/assert.h"
+
+#include <zephyr/bluetooth/classic/rfcomm.h>
+#include <zephyr/bluetooth/classic/hfp_ag.h>
+
+#include "host/hci_core.h"
+#include "host/conn_internal.h"
+#include "l2cap_br_internal.h"
+#include "rfcomm_internal.h"
+#include "at.h"
+#include "sco_internal.h"
+
+#include "hfp_internal.h"
+#include "hfp_ag_internal.h"
+
+#define LOG_LEVEL CONFIG_BT_HFP_AG_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_hfp_ag);
+
+typedef int (*bt_hfp_ag_parse_command_t)(struct bt_hfp_ag *ag, struct net_buf *buf,
+					 enum at_cme *err);
+
+struct bt_hfp_ag_at_cmd_handler {
+	char *cmd;
+	bt_hfp_ag_parse_command_t handler;
+};
+
+static const struct {
+	char *name;
+	char *connector;
```

Comment:
Minor: `const char *` would probably be more appropriate for these.

Reply:
updated